notary-integration-test/src/integration-test/kotlin/integration/btc/BtcWithdrawalIntegrationTest.kt

```diff
@@ -162,6 +168,73 @@ class BtcWithdrawalIntegrationTest {
+        val amount = integrationHelper.btcToSat(2)
+        val randomNameSrc = String.getRandomString(9)
+        val testClientDestKeypair = ModelCrypto().generateKeypair()
+        val testClientSrc = "$randomNameSrc@$CLIENT_DOMAIN"
+        val btcAddressSrc = integrationHelper.registerBtcAddressNoPreGen(randomNameSrc, testClientDestKeypair)
+        integrationHelper.sendBtc(btcAddressSrc, 1, btcWithdrawalConfig.bitcoin.confidenceLevel)
+        integrationHelper.sendBtc(btcAddressSrc, 1, btcWithdrawalConfig.bitcoin.confidenceLevel)
+        val randomNameDest = String.getRandomString(9)
+        val btcAddressDest = integrationHelper.registerBtcAddressNoPreGen(randomNameDest)
+        integrationHelper.addIrohaAssetTo(testClientSrc, btcAsset, amount.toString())
+        integrationHelper.transferAssetIrohaFromClient(
+            testClientSrc,
+            testClientDestKeypair,
+            testClientSrc,
+            btcWithdrawalConfig.withdrawalCredential.accountId,
+            btcAsset,
+            btcAddressDest,
+            amount.toString()
+        )
+        Thread.sleep(WITHDRAWAL_WAIT_MILLIS)
+        assertEquals(initTxCount, createdTransactions.size)
+        transactionHelper.addToBlackList(btcAddressSrc)
+        transactionHelper.addToBlackList(btcAddressDest)
+    }
+
+    /**
+     * Note: Iroha and bitcoind must be deployed to pass the test.
+     * @given two registered BTC clients. 1st client has 2 BTC in wallet
+     * @when 1st client sends 2 BTC to 2nd client without having enough confirmations
+     * @then no tx is created, because nobody can use unconfirmed money
+     */
+    @Test
+    fun testWithdrawalNoConfirmedMoney() {
+        val initTxCount = createdTransactions.size
+        val amount = integrationHelper.btcToSat(2)
+        val randomNameSrc = String.getRandomString(9)
+        val testClientDestKeypair = ModelCrypto().generateKeypair()
+        val testClientSrc = "$randomNameSrc@$CLIENT_DOMAIN"
+        val btcAddressSrc = integrationHelper.registerBtcAddressNoPreGen(randomNameSrc, testClientDestKeypair)
+        integrationHelper.sendBtc(btcAddressSrc, 1, btcWithdrawalConfig.bitcoin.confidenceLevel - 1)
+        val randomNameDest = String.getRandomString(9)
+        val btcAddressDest = integrationHelper.registerBtcAddressNoPreGen(randomNameDest)
+        integrationHelper.addIrohaAssetTo(testClientSrc, btcAsset, amount.toString())
+        integrationHelper.transferAssetIrohaFromClient(
+            testClientSrc,
+            testClientDestKeypair,
+            testClientSrc,
+            btcWithdrawalConfig.withdrawalCredential.accountId,
+            btcAsset,
+            btcAddressDest,
+            amount.toString()
+        )
+        Thread.sleep(WITHDRAWAL_WAIT_MILLIS)
+        assertEquals(initTxCount, createdTransactions.size)
+        transactionHelper.addToBlackList(btcAddressSrc)
+        transactionHelper.addToBlackList(btcAddressDest)
+    }
+
```

Comment:
Also, add test case when spending is possible, but control the fee to be in some limit

Reply:
Can you please add some details? I cannot understand this test case.

Reply:
Maybe some unit cases in which you expect to take specific outputs